packages/realm/src/app-services/SyncConfiguration.ts

```diff
@@ -113,3 +114,26 @@ export function toBindingSyncConfig(config: SyncConfiguration): binding.SyncConf
       : binding.SyncSessionStopPolicy.AfterChangesUploaded,
   };
 }
+
+/** @internal */
+function validatePartitionValue(partitionValue: unknown) {
+  if (partitionValue === undefined) {
+    throw new Error(partitionValue + " is not an allowed PartitionValue");
+  }
+  if (typeof partitionValue == "number") {
```

Comment:
```suggestion
  if (typeof partitionValue === "number") {
```
In general, I think `===` should be used instead of `==`, unless you have a reason to do so. `==` has so many edge cases, that it's possible to get some false positives if you don't know what it's doing.
In this case, it doesn't make much of a difference, as `typeof` will always return a string and `"number"` is also always going to be a string.  
For more on the matter: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

Reply:
hehe I think it was mostly my muscle memory that went for the `==`, I've also heard rumours about `==` being iffy in JS so I'll change it!